src/components/TaskEditor.vue

```diff
@@ -41,6 +41,7 @@
           v-on:update="updateHtmlNotes($event, taskEditorSectionIdAndTask)"
         />
       </div>
+      <DateSelector :date="taskEditorSectionIdAndTask.task.due_on ?? ''" @custom-change="setSelectedDueDate"></DateSelector>
```

Comment:
I think we should just remove the required aspect of the date. Handle the null or undefined in the component directly